Sources/ClientRuntime/PrimitiveTypeExtensions/Indirect.swift

```diff
@@ -5,12 +5,28 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+import class Foundation.NSRecursiveLock
+
 @propertyWrapper
-public class Indirect<T> {
-    public var wrappedValue: Optional<T>
+public final class Indirect<T: Sendable>: @unchecked Sendable {
+    private let lock = NSRecursiveLock()
+    private var _wrappedValue: Optional<T>
+
+    public var wrappedValue: Optional<T> {
+        get {
+            lock.lock()
+            defer { lock.unlock() }
+            return _wrappedValue
+        }
+        set {
+            lock.lock()
+            defer { lock.unlock() }
+            _wrappedValue = newValue
+        }
+    }
```

Comment:
This property wrapper is mostly same as before, but now stores its value in private storage that is protected by a lock when getting & setting.

Since it enforces exclusive access to its wrapped value, it can now be `Sendable`.